docs/reference/speed_python.rst

```diff
@@ -80,6 +80,20 @@ point to sections of the code where performance is not paramount. For example,
 capture ADC readings as integers values to an array in one quick go, and only then
 convert them to floating-point numbers for signal processing.
 
+Integers
+~~~~~~~~
+
+On most ports, integers between -2**30 and 2**30-1 are treated as "small
+integers", and the value is stored directly in the object handle.  For integers
+outside this range, the value is stored in space allocated from the heap,
+resulting in heap allocations, and consequent garbage collection, when such
+values are instantiated.  This occurs where values are calculated from other
+values, or converted from other formats such as an array.
+
+Where possible, working with 32-bit values as pairs of 16-bit values will be
```

Comment:
I honestly don't know what this means, let alone how to apply it. Like: are you talking about building with different integer representation? Or some kind of runtime trick?

Reply:
I agree to @stinos that this sentence `Where possible, ....` is not helpful at all. To explain the situations of allocation and non-allocation, it might be better to include examples like those given by @jimmo. The reference to viper code looks good.

Reply:
I agree that this wasn't clear, and I'm happy to remove it, or expand it to make it clearer.  

The specific situation I was thinking of is where the 32-bit value doesn't represent a single integer, but is actually multiple 8 or 16 bit ints, or is a bitmap of flags and smaller ints, so it's easy to work with the values as separate 8 or 16 bit ints.

Maybe this point is sufficiently obvious (once you understand the limit on small ints) that it's not worth making.

I'm happy to include @jimmo's examples.  Is it worth mentioning that the final one may change in the future? (if/when #9531 is merged)

```
d = a - a  # allocates, even though the result is a small int (zero)
```